### DISCUSSION TITLE
CPS Contract Deployment on Local ICON Node

### DISCUSSION THREAD

**Discussion by aceraizel** [icon-project/btp#31] [Q&A] [answered]
Hi @BennyOptions , I have tried to deploy this CPF to the local node (build from [here](https://github.com/icon-project/goloop/blob/master/doc/gochain_icon_local_node_guide.md))

```
git clone https://github.com/ibriz/CPS/tree/master
cd CPS/CPFTreasury
zip -r -v ./cpf_treasury.zip cpf_treasury lib package.json
cp ./cpf_treasury.zip ${GOCHAIN_LOCAL_ROOT}/

cd ${GOCHAIN_LOCAL_ROOT}
./goloop rpc sendtx deploy ./cpf_treasury.zip --uri http://localhost:9082/api/v3 --key_store ./data/godWallet.json --key_password gochain --nid 3 --step_limit 10000000000 --content_type application/zip
```
Then I have received this error
![image](https://user-images.githubusercontent.com/45514397/114992516-0b48d300-9ec5-11eb-8d3e-9a126b957ff5.png)

Am I missing something? Could you please give me the guide to deploying this contract?

**Comment by ajaya-man** [comment, marked as answer]
Hi @aceraizel,

The error message doesn't help much. Can you let me know, how can I reproduce this locally?

**Reply by aceraizel** [reply]
this local node was build from https://github.com/icon-project/goloop/blob/master/doc/gochain_icon_local_node_guide.md

**Reply by aceraizel** [reply]
@ajaya-man Do you have deployed this CPF contract to the local node?

**Reply by ajaya-man** [reply]
```
# github.com/icon-project/goloop/icon/iiss/icutils
icon/iiss/icutils/utils.go:57:29: syntax error: unexpected _000_000_000_000_000_000, expecting comma or )
note: module requires Go 1.13
# github.com/icon-project/goloop/cmd/cli
cmd/cli/rpc.go:466:41: expireTime.Sub(time.Now()).Microseconds undefined (type time.Duration has no field or method Microseconds)
note: module requires Go 1.13
Makefile:51: recipe for target 'goloop' failed
make: *** [goloop] Error 2
```

I am getting the above error while building the goloop

**Reply by ajaya-man** [reply]
I guess I need a local installation of go version 1.13

**Reply by ajaya-man** [reply]
Hi @aceraizel,

I tried to deploy locally and there was no issue on my side.
```
./goloop rpc sendtx deploy ./CPFTreasury.zip --uri http://localhost:9082/api/v3 --key_store ./data/godWallet.json --key_password gochain --nid 3 --step_limit 10000000000 --content_type application/zip 
"0x797d14560b2e025a5bff702c06f2da9d5d23250508c869597d75e7e6d316581c"```

**Reply by ajaya-man** [reply]
I only zipped the CPF treasury folder.

**Reply by aceraizel** [reply]
@ajaya-man Do you check the result of this transaction hash `"0x797d14560b2e025a5bff702c06f2da9d5d23250508c869597d75e7e6d316581c"` via `./goloop rpc txresult` yet?

**Reply by ajaya-man** [reply]
Yeah, I had checked. It was a success. I can post a screenshot.
![image](https://user-images.githubusercontent.com/16894496/115849605-c385f680-a444-11eb-809d-00ac4f6a4231.png)

**Reply by aceraizel** [reply]
@ajaya-man Can you give me your command to zip the CPF treasury folder?

**Reply by ajaya-man** [reply]
```zip -r ./CPFTreasury.zip CPFTreasury```

CPFTreasury is in the root folder of CPS project.

**Reply by aceraizel** [reply]
@ajaya-man I got successful deployment. Thanks for your support!
